### PR TITLE
Issue# 66 - Summary page updates

### PR DESCRIPTION
**Pull request by reynoldtan**
- Speed up loading of rawdata heatmap summary chart.

### CODE REVIEW

**Comment by reynoldtan** [comment]
**Issue# 66 - Summary page updates**
- Speed up page load time of summary chart. Currently, the summary chart takes 2 mins more or less to render 403K+ data points. With more and more data coming into the system, a significant delay can be observed when accessing summary page.

## Metadata
[https://github.com/UofS-Pulse-Binfo/rawphenotypes/issues/66]

- [X] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [PR #/ nothing]

Documentation:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Description
Summary page references a number of table to generate a summary of traits required by the heatmap. Each trait is then cross-checked for presence of value/data to ensure that it can be visualized. With the growing number of data and process involved to generate a summary, might cause the extended delays.

A table (materialized view) containing a summarized data of all raw phenotypes is available and by using this already summarized information, page load time significantly improved.

## Dependencies
no dependencies.

## Testing?

I compared page load time of summary page in portal versus KP clone and accessed projects available in either projects and compared the set of traits available for visualization. In both trials, page load time with this PR is faster than the portal and number of traits appear to be the same.

Note: This PR now sorts the traits in alphabetical order.

- [ ] I tested on a generic Tripal Site
- [X] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

**Comment by laceysanderson** [comment]
Also, tests need to pass before functional review.

**Comment by reynoldtan** [comment]
Revised query to speed up summary page. Back to you Lacey.

File: include/rawpheno.rawdata.form.inc [hunk 1]
```diff
@@ -43,38 +43,37 @@ function rawpheno_rawdata($form, &$form_state) {
...
+          LIMIT 1";
+
+  // Traits are all in cvterm_ids, we need the human-readable.
+  // Note: query above may return duplicate trait ids since multiple locations share same traits
+  //       but this query ensures that traits are unique.
+  $sql_cvterm_names = "SELECT cvterm_id, name
+          FROM {cvterm}
+          WHERE cvterm_id in (:traits) AND name NOT IN ('Rep', 'Entry', 'Location', 'Name', 'Plot', 'Planting Date (date)', '# of Seeds Planted (count)')
+          GROUP BY cvterm_id, name ORDER BY name ASC";
+
```
[the remark this answers is not in the document]
Comment (laceysanderson) [reply include/rawpheno.rawdata.form.inc]: Updated query looks pretty good :-) However, you should always be wary of `LIMIT 1`. Also, with speed in mind, I find it's faster to order small datasets using PHP then postgresql so you may want to try that as well.

**Pull request merged**